arch/arc/core/reset.S

```diff
@@ -109,6 +112,12 @@ done_icache_invalidate:
 
 done_dcache_invalidate:
 
+#ifdef CONFIG_SOC_ARC_HSDK4XD
+	_soc_early_tune
```

Comment:
I'd probably even make it more detailed. I.e.: `soc_early_init_percpu` or `soc_early_asm_init_percpu`